pkg/operator/staticpod/controller/startupmonitorcondition/startup_monitor_condition.go

```diff
@@ -111,71 +115,70 @@ func (fd *startupMonitorPodConditionController) sync(ctx context.Context, _ fact
+func withDegradedWhenHasFailedContainer(monitorPod *corev1.Pod, startupPodDegraded *applyoperatorv1.OperatorConditionApplyConfiguration) {
 	if monitorPod.Status.Phase != corev1.PodFailed {
 		return
 	}
-	startupPodDegraded.Status = operatorv1.ConditionTrue
-	startupPodDegraded.Reason = "Unknown"
-	startupPodDegraded.Message = "unknown"
+	startupPodDegraded = startupPodDegraded.WithStatus(operatorv1.ConditionTrue).
+		WithReason("Unknown").
+		WithMessage("unknown")
 
 	for _, containerStatus := range monitorPod.Status.ContainerStatuses {
 		if containerStatus.State.Terminated != nil && containerStatus.State.Terminated.ExitCode != 0 {
-			failedContainerReason, failedContainerMessage := startupPodDegraded.Reason, startupPodDegraded.Message
 			if len(containerStatus.State.Terminated.Reason) > 0 {
-				failedContainerReason = containerStatus.State.Terminated.Reason
+				startupPodDegraded.WithReason(containerStatus.State.Terminated.Reason)
+
 			}
 			if len(containerStatus.State.Terminated.Message) > 0 {
-				failedContainerMessage = containerStatus.State.Terminated.Message
+				msg := fmt.Sprintf("at least one container %s in %s pod exited with %d (expected nonzero exit code), due to %v", containerStatus.Name, monitorPod.Name, containerStatus.State.Terminated.ExitCode, containerStatus.State.Terminated.Message)
+				startupPodDegraded.WithMessage(msg)
 			}
-			startupPodDegraded.Reason = failedContainerReason
-			startupPodDegraded.Message = fmt.Sprintf("at least one container %s in %s pod exited with %d (expected nonzero exit code), due to %v", containerStatus.Name, monitorPod.Name, containerStatus.State.Terminated.ExitCode, failedContainerMessage)
 			return
 		}
 	}
 }
 
-func withDegradedWhenHasExcessiveRestarts(monitorPod *corev1.Pod, maxRestartCount int, startupPodContainerExcessiveRestartsDegraded *operatorv1.OperatorCondition) {
+func withDegradedWhenHasExcessiveRestarts(monitorPod *corev1.Pod, maxRestartCount int, startupPodContainerExcessiveRestartsDegraded *applyoperatorv1.OperatorConditionApplyConfiguration) {
 	for _, containerStatus := range monitorPod.Status.ContainerStatuses {
 		if int(containerStatus.RestartCount) >= maxRestartCount {
-			startupPodContainerExcessiveRestartsDegraded.Status = operatorv1.ConditionTrue
-			startupPodContainerExcessiveRestartsDegraded.Reason = "ExcessiveRestarts"
-			startupPodContainerExcessiveRestartsDegraded.Message = fmt.Sprintf("at least one container %s in %s pod has restarted %d times, max allowed is %d", containerStatus.Name, monitorPod.Name, containerStatus.RestartCount, maxRestartCount)
+			startupPodContainerExcessiveRestartsDegraded = startupPodContainerExcessiveRestartsDegraded.WithStatus(operatorv1.ConditionTrue).
+				WithReason("ExcessiveRestarts").
+				WithMessage(fmt.Sprintf("at least one container %s in %s pod has restarted %d times, max allowed is %d", containerStatus.Name, monitorPod.Name, containerStatus.RestartCount, maxRestartCount))
 			return
```

Comment:
We'll need to re-work this function to make sure `startupPodContainerExcessiveRestartsDegraded` status is set to false when needed